pages/addItems/index.tsx

```diff
@@ -0,0 +1,190 @@
+import Button from '@core/ui/buttons/Button';
+import ImageFileInput from '@core/ui/inputs/ImageFileInput';
+import TagInput from '@core/ui/inputs/TagInput';
+import { useState } from 'react';
+import { useAddProduct } from '@lib/queries/items.queries';
+
+interface initValue {
+  imgFile: any;
+  name: string;
+  desc: string;
+  price: number;
+  tags: string[];
+}
+
+const INITIAL_VALUES: initValue = {
+  imgFile: null,
+  name: '',
+  desc: '',
+  price: 0,
+  tags: [],
+};
+
+function AddItems() {
+  const [values, setValues] = useState(INITIAL_VALUES);
+  const [initialPreview, setInitialPreview] = useState<string>('');
+  const [isAllValid, setIsAllValid] = useState(false);
+
+  const addProductMutation = useAddProduct();
+
+  const handleChange = (name: string, value: any) => {
+    setValues((prevValues) => ({
+      ...prevValues,
+      [name]: value,
+    }));
+
+    checkAllInputValid();
+  };
+
+  const checkAllInputValid = () => {
+    if (values.name && values.desc && values.price && values.tags.length > 0) {
+      setIsAllValid(true);
+    } else {
+      setIsAllValid(false);
+    }
+  };
```

Comment:
```suggestion
  const [initialPreview, setInitialPreview] = useState<string>('');

  const isAllValid = (values.name && values.desc && values.price && values.tags.length > 0) 
  const addProductMutation = useAddProduct();

  const handleChange = (name: string, value: any) => {
    setValues((prevValues) => ({
      ...prevValues,
      [name]: value,
    }));
  };
```

handleChange가 되면서 매번 setIsAllvalid를 해주어서 불필요하게 state를 사용하기보단 합쳐서 만드는 것이 코드리딩에 더 좋을 것같아요!

Reply:
가독성이 더 좋아지고 불필요한 상태 업데이트를 하지 않을 수 있을 것 같습니다
수정 완료했습니다 감사합니다!